src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -156,6 +161,11 @@ public class HeaderDecoder : IRlpValueDecoder<BlockHeader>, IRlpStreamDecoder<Bl
                 blockHeader.BaseFeePerGas = rlpStream.DecodeUInt256();
             }
 
+            if (rlpStream.PeekPrefixAndContentLength().ContentLength == Keccak.Size)
```

Comment:
Is it possible that this PeekPrefixAndContentLength() will fail beacuse of https://github.com/NethermindEth/nethermind/blob/9e6ee938062eb536b89bf3a1d43ed7bc6613c43e/src/Nethermind/Nethermind.Serialization.Rlp/Rlp.cs#L841